In [1]:
# Dependencies and Setup
import pandas as pd
import numpy as np

# File to Load 
file_to_load = "Resources/purchase_data.csv"

# Read Purchasing File and store into Pandas data frame
purchase_data = pd.read_csv(file_to_load)

In [2]:
#Provides the first 5 rows of the data file 
purchase_data.head()

Purchase ID             SN  Age Gender  Item ID  \
0            0        Lisim78   20   Male      108   
1            1    Lisovynya38   40   Male      143   
2            2     Ithergue48   24   Male       92   
3            3  Chamassasya86   24   Male      100   
4            4      Iskosia90   23   Male      131   

                                   Item Name  Price  
0  Extraction, Quickblade Of Trembling Hands   3.53  
1                          Frenzied Scimitar   1.56  
2                               Final Critic   4.88  
3                                Blindscythe   3.27  
4                                       Fury   1.44

In [3]:
#Provides simple mathematical analysis of data set, useful for double checking values later on
purchase_data.describe()

Purchase ID         Age     Item ID       Price
count   780.000000  780.000000  780.000000  780.000000
mean    389.500000   22.714103   92.114103    3.050987
std     225.310896    6.659444   52.775943    1.169549
min       0.000000    7.000000    0.000000    1.000000
25%     194.750000   20.000000   48.000000    1.980000
50%     389.500000   22.000000   93.000000    3.150000
75%     584.250000   25.000000  139.000000    4.080000
max     779.000000   45.000000  183.000000    4.990000

## Player Count

* Displays the total number of unique players


In [4]:
#Counting unique users by groupy by function
user_count = purchase_data.groupby("SN")["SN"].nunique().count()

#Creating basic table to display total unique users
user_total_table = pd.DataFrame([{"Total Unique Users":user_count}])
user_total_table

Total Unique Users
0                 576

## Purchasing Analysis (Total)

* Run basic calculations to obtain number of unique items, average price, etc.


* Create a summary data frame to hold the results


* Optional: give the displayed data cleaner formatting


* Display the summary data frame


In [5]:
#Counts unique items by grouping unique item names
unique_items = purchase_data.groupby("Item ID")["Item ID"].nunique().count()

#Takes average value of the price column
avg_price = purchase_data["Price"].mean()

#Adds together all values in prices coliumn to provide total revenue
tot_sales = purchase_data["Price"].sum()

#Counts the number of rows in the data set to show total number of purchases
num_purchase = purchase_data.shape[0]

#Creating dictionary to assign to new dataframe, makes reading summary table easier
pd.options.display.float_format = '${:,.2f}'.format
summary_table = pd.DataFrame([{"Number of Unique Items": unique_items,"Average Price": round(avg_price,2),"Number of Purchases": num_purchase,"Total Revenue": round(tot_sales,2)}])
summary_table

Number of Unique Items  Average Price  Number of Purchases  Total Revenue
0                     183          $3.05                  780      $2,379.77

## Gender Demographics

* Percentage and Count of Male Players


* Percentage and Count of Female Players


* Percentage and Count of Other / Non-Disclosed




In [6]:
#Grouping by gender and defining percentage of which genders are playing
genders_df = purchase_data.groupby("Gender")["SN"].nunique()

#Calculating percentage of players by gender
genders_perc_df = (genders_df/576)*100
genders_perc_df.round(2)

#Creating table to hold all relevant information
pd.options.display.float_format = '{:,.2f}%'.format
gender_demo_table = pd.DataFrame({"Gender Count":genders_df , "Gender Percentage":round(genders_perc_df,2)})
gender_demo_table

Gender Count  Gender Percentage
Gender                                                
Female                           81             14.06%
Male                            484             84.03%
Other / Non-Disclosed            11              1.91%


## Purchasing Analysis (Gender)

* Run basic calculations to obtain purchase count, avg. purchase price, avg. purchase total per person etc. by gender




* Create a summary data frame to hold the results


* Optional: give the displayed data cleaner formatting


* Display the summary data frame

In [7]:
#Grouping purchase count by gender 
gender_purchases = purchase_data.groupby("Gender")["Item Name"].count()

#Finding average price of purchases grouped by gender rounded to 2 decimals
gender_avg = purchase_data.groupby("Gender")["Price"].mean()
gender_avg.round(2)

#Total spent on items grouped by gender
gender_total = purchase_data.groupby("Gender")["Price"].sum()
gender_total.round(2)

#Total spent per person from each gender
per_person_total = gender_total/genders_df
per_person_total.round(2)

#Creating the display table and applying formatting to make it all prettier
pd.options.display.float_format = '${:,.2f}'.format
gender_purch_table = pd.DataFrame({"Purchase Count":gender_purchases, "Average Purchase Price":round(gender_avg,2),"Total Purchase Value":gender_total, "Avg. Total Purchase per Person":round(per_person_total,2)})
gender_purch_table

Purchase Count  Average Purchase Price  \
Gender                                                          
Female                            113                   $3.20   
Male                              652                   $3.02   
Other / Non-Disclosed              15                   $3.35   

                       Total Purchase Value  Avg. Total Purchase per Person  
Gender                                                                       
Female                              $361.94                           $4.47  
Male                              $1,967.64                           $4.07  
Other / Non-Disclosed                $50.19                           $4.56

## Age Demographics

* Establish bins for ages


* Categorize the existing players using the age bins. Hint: use pd.cut()


* Calculate the numbers and percentages by age group


* Create a summary data frame to hold the results


* Optional: round the percentage column to two decimal points


* Display Age Demographics Table


In [8]:
# Establish bins for ages
age_bins = [0, 9.99, 14.99, 19.99, 24.99, 29.99, 34.99, 39.99, 99999]
group_names = ["<10", "10-14", "15-19", "20-24", "25-29", "30-34", "35-39", "40+"]

#Use cut function to place relevant data into bins for analysis
purchase_data["Age Group"] = pd.cut(purchase_data["Age"],age_bins, labels=group_names)

# Create new data frame with the added "Age Group" and group it
age_grouped = purchase_data.groupby("Age Group")

# Count total players by age category
total_count_age = age_grouped["SN"].nunique()

# Calculate percentages by age category 
percentage_by_age = (total_count_age/user_count) * 100

# Create data frame 
pd.options.display.float_format = '{:,.2f}%'.format

age_demographics = pd.DataFrame({"Percentage of Players": percentage_by_age, "Total Count": total_count_age})
age_demographics

Percentage of Players  Total Count
Age Group                                    
<10                        2.95%           17
10-14                      3.82%           22
15-19                     18.58%          107
20-24                     44.79%          258
25-29                     13.37%           77
30-34                      9.03%           52
35-39                      5.38%           31
40+                        2.08%           12

## Purchasing Analysis (Age)

* Bin the purchase_data data frame by age


* Run basic calculations to obtain purchase count, avg. purchase price, avg. purchase total per person etc. in the table below


* Create a summary data frame to hold the results


* Optional: give the displayed data cleaner formatting


* Display the summary data frame

In [9]:
# Count purchases by age group
purchase_count_age = age_grouped["Purchase ID"].count()

# Obtain average purchase price by age group 
avg_purchase_price_age = age_grouped["Price"].mean()

# Calculate total purchase value by age group 
total_purchase_value = age_grouped["Price"].sum()

# Calculate the average purchase per person in the age group 
avg_purchase_per_person_age = total_purchase_value/total_count_age

# Create data frame with obtained values and formatting
pd.options.display.float_format = '${:,.2f}'.format

age_demographics_purchase = pd.DataFrame({"Purchase Count": purchase_count_age,
                                 "Average Purchase Price": avg_purchase_price_age,
                                 "Total Purchase Value":total_purchase_value,
                                 "Average Purchase Total per Person": avg_purchase_per_person_age})
age_demographics_purchase

Purchase Count  Average Purchase Price  Total Purchase Value  \
Age Group                                                                 
<10                    23                   $3.35                $77.13   
10-14                  28                   $2.96                $82.78   
15-19                 136                   $3.04               $412.89   
20-24                 365                   $3.05             $1,114.06   
25-29                 101                   $2.90               $293.00   
30-34                  73                   $2.93               $214.00   
35-39                  41                   $3.60               $147.67   
40+                    13                   $2.94                $38.24   

           Average Purchase Total per Person  
Age Group                                     
<10                                    $4.54  
10-14                                  $3.76  
15-19                                  $3.86  
20-24                                  $4.32  
25-29                                  $3.81  
30-34                                  $4.12  
35-39                                  $4.76  
40+                                    $3.19

## Top Spenders

* Run basic calculations to obtain the results in the table below


* Create a summary data frame to hold the results


* Sort the total purchase value column in descending order


* Optional: give the displayed data cleaner formatting


* Display a preview of the summary data frame



In [10]:
# Spending analysis for each individual user which will then be sorted by top spender
players_purchase_count_df = purchase_data.groupby("SN").count()["Price"].rename("Purchase Count")
players_average_price_df = purchase_data.groupby("SN").mean()["Price"].rename("Average Purchase Price")
players_total_df = purchase_data.groupby("SN").sum()["Price"].rename("Total Purchase Value")

#Convert to DataFrame.
total_user_data_df = pd.DataFrame({"Purchase Count":players_purchase_count_df,
                                   "Average Purchase Price": players_average_price_df,
                                   "Total Purchase Value": players_total_df})

#Sorting by top spenders and displaying the top 5, starting with the biggest spender
top_five_spenders = total_user_data_df.sort_values("Total Purchase Value", ascending=False)
top_five_spenders.head()


Purchase Count  Average Purchase Price  Total Purchase Value
SN                                                                       
Lisosia93                 5                   $3.79                $18.96
Idastidru52               4                   $3.86                $15.45
Chamjask73                3                   $4.61                $13.83
Iral74                    4                   $3.40                $13.62
Iskadarya95               3                   $4.37                $13.10

## Most Popular Items

* Retrieve the Item ID, Item Name, and Item Price columns


* Group by Item ID and Item Name. Perform calculations to obtain purchase count, item price, and total purchase value


* Create a summary data frame to hold the results


* Sort the purchase count column in descending order


* Optional: give the displayed data cleaner formatting


* Display a preview of the summary data frame



In [11]:
# Total items purchased analysis.
items_purchase_count_df = purchase_data.groupby(["Item ID", "Item Name"]).count()["Price"].rename("Purchase Count")
items_average_price_df = purchase_data.groupby(["Item ID", "Item Name"]).mean()["Price"].rename("Average Purchase Price")
items_value_total_df = purchase_data.groupby(["Item ID", "Item Name"]).sum()["Price"].rename("Total Purchase Value")

# Convert to DataFrame
items_purchased_df = pd.DataFrame({"Purchase Count":items_purchase_count_df,
                                   "Item Price":items_average_price_df,
                                   "Total Purchase Value":items_value_total_df,})

#Sorting by most popular items, and displaying the top 5 results
most_popular_items_df = items_purchased_df.sort_values("Purchase Count", ascending=False)
most_popular_items_df.head()

Purchase Count  \
Item ID Item Name                                                      
178     Oathbreaker, Last Hope of the Breaking Storm              12   
145     Fiery Glass Crusader                                       9   
108     Extraction, Quickblade Of Trembling Hands                  9   
82      Nirvana                                                    9   
19      Pursuit, Cudgel of Necromancy                              8   

                                                      Item Price  \
Item ID Item Name                                                  
178     Oathbreaker, Last Hope of the Breaking Storm       $4.23   
145     Fiery Glass Crusader                               $4.58   
108     Extraction, Quickblade Of Trembling Hands          $3.53   
82      Nirvana                                            $4.90   
19      Pursuit, Cudgel of Necromancy                      $1.02   

                                                      Total Purchase Value  
Item ID Item Name                                                           
178     Oathbreaker, Last Hope of the Breaking Storm                $50.76  
145     Fiery Glass Crusader                                        $41.22  
108     Extraction, Quickblade Of Trembling Hands                   $31.77  
82      Nirvana                                                     $44.10  
19      Pursuit, Cudgel of Necromancy                                $8.16

## Most Profitable Items


* Sort the above table by total purchase value in descending order


* Optional: give the displayed data cleaner formatting


* Display a preview of the data frame



In [12]:
#Sorting by most profitable items, and displaying the top 5 results
most_profitable_items_df = items_purchased_df.sort_values("Total Purchase Value", ascending=False)
most_profitable_items_df.head()

Purchase Count  \
Item ID Item Name                                                      
178     Oathbreaker, Last Hope of the Breaking Storm              12   
82      Nirvana                                                    9   
145     Fiery Glass Crusader                                       9   
92      Final Critic                                               8   
103     Singed Scalpel                                             8   

                                                      Item Price  \
Item ID Item Name                                                  
178     Oathbreaker, Last Hope of the Breaking Storm       $4.23   
82      Nirvana                                            $4.90   
145     Fiery Glass Crusader                               $4.58   
92      Final Critic                                       $4.88   
103     Singed Scalpel                                     $4.35   

                                                      Total Purchase Value  
Item ID Item Name                                                           
178     Oathbreaker, Last Hope of the Breaking Storm                $50.76  
82      Nirvana                                                     $44.10  
145     Fiery Glass Crusader                                        $41.22  
92      Final Critic                                                $39.04  
103     Singed Scalpel                                              $34.80

# Written Analysis

Their are 576 unique users Playerbase is primarily in their teens to late 20's. Purchase analysis shows that players in the age range 35-39 and <10 have higher average spending per transaction. To increase profitability consider marketing to the older or younger audience to increase profits and grow the community. Potentially develop a new game mode designed for young kids to increase total <10 playerbase. The gender discrepancy is large, males account for 84% of the player base. Market to women, or create characters that will excite and attract a larger female audience.

Oathbreaker, Last Hope of the Breaking Storm is the most popular and profitable item. Pursuit, Cudgel of Necromancy, is in the top 5 most popular items list, consider increasing the price to increase profitability. Lisosia93 is our biggest spender, send them a message or gift them a unique item as a thanks for their continued loyalty to encourage other players to spend more. Consider doing this for the top 5 spenders every month to increase competition in the community, driving up profits.